ui/dashboards/src/test/render.tsx

```diff
@@ -22,3 +26,25 @@ export function renderWithContext(ui: React.ReactElement, options?: Omit<RenderO
   const queryClient = new QueryClient({ defaultOptions: { queries: { refetchOnWindowFocus: false, retry: false } } });
   return render(<QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>, options);
 }
+
+/**
+ * Similar to renderWithContext test helper but to test routing / query param functionality
+ * see: https://github.com/pbeshai/use-query-params/blob/master/packages/use-query-params-adapter-react-router-6/src/__tests__/react-router-6.test.tsx
+ */
+export function renderWithHistory(ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) {
```

Comment:
If you wanted to combine this into `renderWithContext`, I wouldn't be opposed. Our internal test helper has an option for initial `history` state because it's so common of a need.